pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -410,6 +412,12 @@ func CreateGCECloud(config *CloudConfig) (*GCECloud, error) {
 		kmsConfig:                config.KMSConfig,
 	}
 
+	// Register the provided KMS service with the plugin registry to allow using
+	// it for encryption of resources.
+	encryptionconfig.KMSPluginRegistry.Register(KMSServiceName, func(_ io.Reader) (envelope.Service, error) {
```

Comment:
since InitCloudProvider can be called multiple times, we need to ensure that this works if multiple are registered for the same plugin name. that probably means the registered plugin must be stateless